scripts/uosc_shared/lib/utils.lua

```diff
@@ -164,10 +181,31 @@ function normalize_path(path)
 		end
 		return path
 	else
-		return trim_end(path, '/')
+		path = trim_end(path, path:match('^\\\\') and '\\' or '/')
+		if path == '' then path = '/' end
+		return path
 	end
```

Comment:
Extract this if/else into `trim_trailing_separator()` function so that it can be used in `super_normalize_path()` instead of calling `normalize_path()` there, which does a lot of duplicitous work.

Also, wouldn't `path:sub(1, 2) == '\\\\'` be faster than `:match('^\\\\')`?

Reply:
I've timed it and `:find()` is faster then `:match()` in theory because it doesn't have to create a string for returning (when there are no captures), but I couldn't really measure a difference in our case.
`:sub()` did bring a small but measurable improvement.